### PR TITLE
Add storage overlays for IBM NFS

### PR DESCRIPTION
**Pull request by Tomcli**
**Which issue is resolved by this Pull Request:**
Related issue: https://github.com/kubeflow/katib/issues/725

**Description of your changes:**
The IBM Cloud NFS(Network File System) on IBM's Kubernetes service is different from regular File System storage, and we encounter several issues with some of the mysql and oidc storage setup. Therefore, we created some overlays to config to the settings that can run on IBM NFS. 

cc: @animeshsingh 

**Checklist:**
- [x] Unit tests have been rebuilt: 
    1. `cd manifests/tests`
    2. `make generate`
    3. `make test`

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/manifests/622)

### CODE REVIEW

**Comment by Tomcli** [comment]
Looks like the util library for `build_kfctl_go()` is updated, but the testing script is not up to date. https://github.com/kubeflow/kubeflow/blob/master/testing/kfctl/kfctl_go_test.py#L38

Let me submit an issue for this.

**Comment by animeshsingh** [comment]
@krishnadurai @kkasravi can you please take a look. These are customizations needed to use this on IBM Cloud Storage configuration

**Comment by kunmingg** [comment]
Shall we use one DB deployment for both metadata and katib?

**Comment by kunmingg** [comment]
GCP and AWS have entry like 
https://github.com/kubeflow/manifests/tree/master/gcp
https://github.com/kubeflow/manifests/tree/master/aws
for custom components.
An alternative is to create a 
https://github.com/kubeflow/manifests/tree/master/ibm

**Comment by Tomcli** [comment]
Hi @kunmingg, we have custom overlays for the metadata db and katib db. Is there a way for us to apply those overlays by putting our overlay code under https://github.com/kubeflow/manifests/tree/master/ibm?

**Comment by krishnadurai** [comment]
@animeshsingh @Tomcli I will take a look at this later today.

File: istio/oidc-authservice/overlays/ibm-storage-config/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: authservice
+spec:
+  template:
+    spec:
+      initContainers:
+      - name: fix-permission
+        image: busybox
```
Comment (krishnadurai) [inline istio/oidc-authservice/overlays/ibm-storage-config/deployment.yaml]: Is this image referenced in kustomization.yaml for the overlay? We do this to ensure that a different container repository can be used.

File: katib/katib-controller/overlays/ibm-storage-config/katib-db-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: katib-db
+spec:
+  template:
+    spec:
+      containers:
+      - name: katib-db
+        livenessProbe: null
```
Comment (krishnadurai) [inline katib/katib-controller/overlays/ibm-storage-config/katib-db-deployment.yaml]: A comment explaining why livenessProbe is disabled would be helpful.

File: katib/katib-controller/overlays/ibm-storage-config/kustomization.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+bases:
+- ../../base
+patchesStrategicMerge:
+- katib-db-deployment.yaml
+images:
+  - name: mysql
+    newTag: "5.6"
```
Comment (krishnadurai) [inline katib/katib-controller/overlays/ibm-storage-config/kustomization.yaml]: Is 5.6 feature complete for Katib? The default seems to be at 8.

File: metadata/overlays/ibm-storage-config/kustomization.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+bases:
+- ../../base
+images:
+  - name: mysql
+    newTag: "5.6"
```
Comment (krishnadurai) [inline metadata/overlays/ibm-storage-config/kustomization.yaml]: Same here. Is the DB version sufficient?

File: istio/oidc-authservice/overlays/ibm-storage-config/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+        command: ['sh', '-c']
+        args: ['chmod -R 777 /var/lib/authservice;']
```
[the remark this answers is not in the document]
Comment (Tomcli) [reply istio/oidc-authservice/overlays/ibm-storage-config/deployment.yaml]: We are getting some permission errors with 755, so we changed it to 777. Below are the errors we had if we set it as `chmod -R 755 /var/lib/authservice`
```
time="2019-11-27T19:23:22Z" level=info msg="Starting readiness probe at 8081"
time="2019-11-27T19:23:22Z" level=info msg="No  USERID_TOKEN_HEADER  specified, using 'kubeflow-userid-token' as default."
time="2019-11-27T19:23:22Z" level=info msg="No  SESSION_MAX_AGE  specified, using '86400' as default."
time="2019-11-27T19:23:22Z" level=fatal msg="Error opening bolt store: open /var/lib/authservice/data.db: permission denied"
```

File: istio/oidc-authservice/overlays/ibm-storage-config/deployment.yaml (hunk 1 above)
Comment (Tomcli) [reply istio/oidc-authservice/overlays/ibm-storage-config/deployment.yaml]: I committed the below section in the kustomization.yaml. We should able to change it to other container repository now.
```
images:
  - name: busybox
    newTag: "latest"
    newName: busybox
```

File: katib/katib-controller/overlays/ibm-storage-config/katib-db-deployment.yaml (hunk 2 above)
Comment (Tomcli) [reply katib/katib-controller/overlays/ibm-storage-config/katib-db-deployment.yaml]: I added a comment in my latest commit, we are getting some segmentation fault when the livenessProbe hit our db pod.

File: katib/katib-controller/overlays/ibm-storage-config/kustomization.yaml (hunk 3 above)
Comment (Tomcli) [reply katib/katib-controller/overlays/ibm-storage-config/kustomization.yaml]: We discussed with the Katib team at https://github.com/kubeflow/katib/issues/725 and they said the katib-db didn't use any mysql 8 specific feature.

File: metadata/overlays/ibm-storage-config/kustomization.yaml (hunk 4 above)
Comment (Tomcli) [reply metadata/overlays/ibm-storage-config/kustomization.yaml]: The metadata-db is using mysql to do basic database operations similar to pipeline's mysql. Therefore, we aligned with pipeline's mysql 5.6 and it works smoothly on our environment.

**Comment by krishnadurai** [comment]
@Tomcli looks like my approval might not be enough.

@kunmingg could you please take a look?

**Comment by animeshsingh** [comment]
Thanks @krishnadurai 
@kunmingg pretty important for us to get this going on ibm cloud NFS. Please take a look

**Comment by kunmingg** [comment]
/approve
Sorry was ooo earlier

**Comment by Tomcli** [comment]
Thanks @kunmingg

File: istio/oidc-authservice/overlays/ibm-storage-config/deployment.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (javen218) [reply istio/oidc-authservice/overlays/ibm-storage-config/deployment.yaml:12]: When I was deploying kubeflow, I also met the error "**Error opening bolt store: open /var/lib/authservice/data.db: permission denied**". 

I found change the permission restrictive of pv path is work:  `chmod -R 777 /paasdata/var/data/share/kubeflow/authservice`

But this way is not safe enough, is there any other solutions to solve this problem? @Tomcli 

`[root@paas-controller-2 manifests-1.4.1]# ki logs authservice-0
time="2022-04-07T07:08:02Z" level=info msg="Starting readiness probe at 8081"
time="2022-04-07T07:08:02Z" level=info msg="No  USERID_TOKEN_HEADER  specified, using 'kubeflow-userid-token' as default."
time="2022-04-07T07:08:02Z" level=info msg="No  SERVER_HOSTNAME  specified, using '' as default."
time="2022-04-07T07:08:02Z" level=info msg="No  SERVER_PORT  specified, using '8080' as default."
time="2022-04-07T07:08:02Z" level=info msg="No  SESSION_MAX_AGE  specified, using '86400' as default."
time="2022-04-07T07:08:02Z" level=info msg="Starting web server at :8080"
time="2022-04-07T07:08:02Z" level=fatal msg="Error opening bolt store: open /var/lib/authservice/data.db: permission denied"`

**Pull request merged**